Speed_Controller/src/Timer/Timer.c

```diff
@@ -1,61 +1,82 @@
 #include "Timer.h"
 
-int8_t timeoutFlag = FALSE;
+#define TRUE 1
+#define FALSE 0
 
-/** Interrupt handler for Timer 2.
+static volatile int8_t timeoutFlag = FALSE;
+
+/**
+ * Initializes a timer in one pulse mode to trigger a timing interrupt after a set interval
+ *
+ * Assumes APB2 timer clock of 36MHz
+ * 
+ * @param period Period for triggering the timing interrupt in ms (must be < 65535 / 10)
  */
-void TIM2_IRQHandler(void)
+void TimerInit(unsigned int period)
 {
-	
-	if (TIM2->SR & 0x1){
-		
-		//Set a flag for action here.
-		timeoutFlag = TRUE;
-			
-		TIM2->SR &= ~(0x1);		//reset the update interrupt flag
-	}
-	
+	RCC->APB1ENR |= RCC_APB1ENR_TIM2EN; // Enable TIM2 clock
+
+	TIM2->PSC = TIM2_PRESCALER; // Processor speed is 36MHz; set prescaler value for a clock frequency of 10kHz
+
+	TIM2->CR1 &= ~TIM_CR1_DIR;	// Set counter to upcount
+	TIM2->CR1 &= ~TIM_CR1_CKD; 	// Set the timer to use default clock division
+	TIM2->CR1 |= TIM_CR1_OPM; 	// Set the timer to use one pulse mode
+
+	TIM2->ARR = 10 * period - 1;// Set autoreload value to set the period
+
+	TIM2->DIER |= TIM_DIER_UIE;	// Enable timing interrupt
+	NVIC_EnableIRQ(TIM2_IRQn);	// Setup interrupt handler
+
+	DBGMCU->CR |= DBGMCU_CR_DBG_TIM2_STOP; // Halt TIM2 upon debug halt
+
+	TIM2->CR1 |= TIM_CR1_CEN; 	// Enable TIM2
 }
 
-/** Initializes a timer to trigger a timing interrupt periodically
- * @param period: period for triggering the timing interrupt in ms
+/**
+ * Restarts the counter on the timer
  */
-void TimerInit(int period){
-	
-	RCC->APB1ENR |= 0x1;		//enable TIM2
-	
-	TIM2->PSC = TIM2_PRESCALER;	//Processor speed is ~72MHZ, so set prescaler value for scaling, clock frequency of 10kHz
-	
-	TIM2->CR1 &= ~(0x1UL << 4);	//set counter to upcount
-	
-	TIM2->ARR &= 0;
-	
-	TIM2->ARR = 10*period;		//set autoreload to reset every period
-	
-	TIM2->CR1 &= (0x11UL << 8);	//set the timer to use default clock division
-
-	TIM2->CR1 |= 0x1UL << 3;	//set the timer to use one pulse mode
-	
-	TIM2->CR1 |= 0x1UL;			//enable TIM2
-	
-	TIM2->DIER |= 0x1UL;		//enable timing interrupt	
-	NVIC_EnableIRQ(TIM2_IRQn);	//setup interrupt handler
+void RestartTimer(void)
+{
+	TIM2->CNT = 0;
+	TIM2->CR1 |= TIM_CR1_CEN; // Re-enable TIM2

```

Comment:
Enabling the timer should be the very last step of the initialization. This was corrected.